algebra/src/lib.rs

```diff
@@ -82,3 +97,5 @@ pub mod prelude {
 
     pub use num_traits::{One, Zero};
 }
+
+mod fake_io;
```

Comment:
Can we rename this to `io`, and make it conditionally equal the `std` module:
```suggestion
#[cfg(not(feature = "std"))]
mod io;
#[cfg(feature = "std")]
pub use std::io;
```

Reply:
yes, I think it's doable

Reply:
done